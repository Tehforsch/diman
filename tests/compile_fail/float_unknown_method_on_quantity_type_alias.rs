#![allow(incomplete_features)]
#![feature(generic_const_exprs, adt_const_params)]
pub mod example_system;

fn main() {
    use example_system::f64::Length;
    Length::unknown_method(49.0);
}
