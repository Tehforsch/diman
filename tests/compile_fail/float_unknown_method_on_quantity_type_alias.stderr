error[E0599]: no function or associated item named `unknown_method` found for struct `example_system::Quantity` in the current scope
 --> tests/compile_fail/float_unknown_method_on_quantity_type_alias.rs:7:13
  |
7 |     Length::unknown_method(49.0);
  |             ^^^^^^^^^^^^^^ function or associated item not found in `Quantity<f64, Dimension>`
  |
 ::: tests/compile_fail/example_system/mod.rs
  |
  |     quantity_type Quantity;
  |                   -------- function or associated item `unknown_method` not found for this struct
  |
note: if you're trying to build a new `example_system::Quantity<f64, Dimension>` consider using one of the following associated functions:
      example_system::Quantity::<f64, Dimension>::dimensionless
      example_system::Quantity::<f64, Dimension>::grams
      example_system::Quantity::<f64, Dimension>::joules
      example_system::Quantity::<f64, Dimension>::kilograms
      and $N others
 --> tests/compile_fail/example_system/mod.rs
  |
  |     dimension_type Dimension;
  |                    ^^^^^^^^^
